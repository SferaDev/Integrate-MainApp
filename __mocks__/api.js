const API = {
    Function: () => {
        return new Promise(function(resolve, reject){
            console.log('fgjsdusdg');
            console.log("Soc API");
            resolve("Soc RESOLVE");
        });
    }
}

export default API