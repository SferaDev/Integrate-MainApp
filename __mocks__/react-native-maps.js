import React from 'react';
import {
  View
} from 'react-native';

const mockGeolocation = {
  getCurrentPosition: jest.fn(),
  watchPosition: jest.fn()
};

global.navigator = {};
global.navigator.geolocation = mockGeolocation;

export default class MapView extends React.Component {

  static Marker = '';

  render() {
    return (null);
  }
}