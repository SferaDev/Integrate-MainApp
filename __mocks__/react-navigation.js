import React from 'react';
import { View, Text } from 'react-native';

class ReactNavigation extends React.Component {

	static NavigationContainer(){
		return(<View><Text>HOLA</Text></View>);
	}

	static StackNavigator(){
	 return ReactNavigation.NavigationContainer 
	}

	static DrawerNavigator(){
	 return ReactNavigation.NavigationContainer 
	}

  render() {
    return (null);
  }
}

export default ReactNavigation;