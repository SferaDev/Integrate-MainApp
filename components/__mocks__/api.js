const GOODS = ['111', '222', '333', '444', '555', '666', '777', '888', '999'];
const USED_GOODS = ['111', '222', '333', '777'];

const API = {
    getEntity: (id = null) => {
        return new Promise(async (resolve, reject) => {
            let entity = {
                _id: 0,
                name: 'name',
                description: 'description',
                addressName: 'addressName',
                phone: '000000',
                email: 'aaa@bbb.com',
                coordinates: [0, 0],
                goods: []
            };
            resolve(entity);
        });
    },
    checkOrder: async (selected_goods = []) => {

        let ok = true;
        for (let sg of selected_goods) {
            if (!GOODS.includes(sg) || USED_GOODS.includes(sg)) ok = false;
        }

        if (ok) {
            return {status: 200, body: {total_discount: 0}};
        } else {
            return {status: 409, body: {soldOutGoods: [], nonUsableGoods: []}};
        }
    },
    newOrder: async (selected_goods = [], entityId = null, validationCode = null) => {

        if (entityId === null || validationCode === null) return {status: 403, body: 'Error'};

        let ok = true;
        for (let sg of selected_goods) {
            if (!GOODS.includes(sg) || USED_GOODS.includes(sg)) ok = false;
        }

        if (ok) {
            return {status: 201, body: {total_discount: 0}};
        } else {
            return {status: 409, body: {soldOutGoods: [], nonUsableGoods: []}};
        }
    },
    getLanguages: async () => {

        return [{name: 'Catalan',language: 'ca'}];
    },
};

export default API
