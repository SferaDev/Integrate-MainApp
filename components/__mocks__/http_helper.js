const BASE_URL = 'http://integrate-backend-staging.herokuapp.com';

const buildQuery = (url = '', params = [], base_url = BASE_URL) => {
    let query = base_url + '/' + url;
    let keys = Object.keys(params);

    for (let i in keys) {
        if (i == 0) query += '?';
        else query += '&';
        query += params[keys[i]].key + '=' + params[keys[i]].value;
    }
    return query;
};

const callApi = async (url, params, method = 'GET') => {
    //fetch(buildQuery(url, params), { method: method }).then(success).catch(error);

    let isNull = false;
    for(let p of params){
        if( p.key != "token" && p.value == null )isNull = true;
    }

    if( url.split('/')[url.split('/').length - 1] == "null" || isNull ){
        return await {
            status: url === 'login' ? 401 : 404,
            _bodyText: JSON.stringify({something: 'Not found'})
        };
    }else{
        return await {
            status: 200,
            _bodyText: JSON.stringify({something: 'Hello World'})
        };
    }
};

const httpHelper = {
    buildQuery: buildQuery,
    callApi: callApi
};

export default httpHelper;
