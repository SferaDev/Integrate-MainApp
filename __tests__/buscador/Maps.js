import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import {Marker} from '../../__mocks__/react-native-maps';
import Maps from '../../components/buscador/maps';

let wrapper;
let instance;
let entity = {
            _id: 0,
            name: 'name',
            description: 'description',
            addressName: 'addressName',
            phone: '000000',
            email: 'aaa@bbb.com',
            coordinates: [0, 0],
            goods: []
        }

describe('Test group for Maps', function () {
    beforeAll(() => {

        configure({adapter: new Adapter()});
    });

    beforeEach(function () {
        // Before each: Shallows the Maps component
        wrapper = shallow(<Maps entities={[entity]}/>);
        instance = wrapper.instance();
    });

    afterEach(function () {
        // After each: Clears the wrapper
        wrapper = null;
        instance = null;
    });

    test('renders maps correctly', () => {
        const tree = renderer.create(<Maps entities={[entity]}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('goToMe is callable and returns nothing', () => {
        expect(instance.goToMe()).toBe(undefined);
    });

    test('mapAnimateToMe is callable and returns nothing', () => {
        instance.map = {animateToRegion: jest.fn()};
        expect(instance.mapAnimateToMe({
            coords: {
                latitude: 0,
                longitude: 0,
                latitudeDelta: 0,
                longitudeDelta: 0
            }
        })).toBe(undefined);
    });

    test('pointNorth is callable and returns nothing', () => {
        instance.map = {animateToBearing: jest.fn()};
        expect(instance.pointNorth({
            coords: {
                latitude: 0,
                longitude: 0,
                latitudeDelta: 0,
                longitudeDelta: 0
            }
        })).toBe(undefined);
    });
});
