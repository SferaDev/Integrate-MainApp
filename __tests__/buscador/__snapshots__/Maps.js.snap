// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test group for Maps renders maps correctly 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    followsUserLocation={true}
    initialRegion={
      Object {
        "latitude": 0,
        "latitudeDelta": 0.0922,
        "longitude": 0,
        "longitudeDelta": 0.0421,
      }
    }
    showsCompass={false}
    showsMyLocationButton={false}
    showsPointsOfInterest={false}
    showsUserLocation={true}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "paddingTop": 100,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    toolbarEnabled={false}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      coordinate={
        Object {
          "latitude": 0,
          "latitudeDelta": 0.01,
          "longitude": 0,
          "longitudeDelta": 0.01,
        }
      }
      ellipsizeMode="tail"
      identifier="0"
      image={
        Object {
          "testUri": "../../../Images/marker60.png",
        }
      }
      onPress={[Function]}
      pinColor="red"
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "backgroundColor": "white",
          "borderRadius": 15,
          "bottom": 15,
          "display": "flex",
          "height": 30,
          "justifyContent": "center",
          "left": 15,
          "position": "absolute",
          "width": 30,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          Object {
            "alignSelf": "center",
            "color": "#67ACB1",
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "backgroundColor": "white",
          "borderRadius": 22.5,
          "bottom": 15,
          "display": "flex",
          "height": 45,
          "justifyContent": "center",
          "position": "absolute",
          "right": 15,
          "width": 45,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          Object {
            "alignSelf": "center",
            "color": "#67ACB1",
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
